src/neo/SmartContract/Native/Role.cs

```diff
@@ -18,6 +18,11 @@ public enum Role : byte
         /// <summary>
         /// NeoFS Alphabet nodes.
         /// </summary>
-        NeoFSAlphabetNode = 16
+        NeoFSAlphabetNode = 16,
+
+        /// <summary>
+        /// Notary nodes
+        /// </summary>
+        P2PNotary = 128
```

Comment:
I think that another role only for multi signature exchange it's too much complicated. We should do it without extra nodes.

Reply:
While technically any full node can collect signatures they might as well not do it. And even if some voluntarily do that we don't know which one is to receive the reward. So these designated nodes are economically motivated for performing their job, that's the main reason for this role. And it's yet another proper network service, you pay for using it and the payment goes to this list of beneficiaries.

Reply:
If you do this, why not create a website or web api to handle it? I'm not sure if it is worth doing on the blockchain.

Reply:
The rationale for this service is laid out in #1573, it's exactly to prevent building app-specific third-party solutions for the problem that can be solved in generic way. The blockchain part of it is just deposit management/rewards to prevent abuse and provide economic stimulation for nodes providing this service, otherwise it's purely a P2P protocol thing, the chain just gets the resulting complete transaction.